src/fs/inotify.rs

```diff
@@ -24,12 +24,12 @@ pub fn inotify_init(flags: CreateFlags) -> io::Result<OwnedFd> {
 /// this method may result in it returning the same watch descriptor. An
 /// application should keep track of this externally to avoid logic errors.
 #[inline]
-pub fn inotify_add_watch<P: crate::path::Arg>(
-    inot: BorrowedFd<'_>,
+pub fn inotify_add_watch<Fd: AsFd, P: crate::path::Arg>(
```

Comment:
Actually I think adding a generic isn't backwards compatible if someone explicitly specified `P`. Then again, I don't think anybody uses this code? https://github.com/search?q=inotify_add_watch+language%3ARust+rustix+-is%3Afork&type=code At least not publicly.

Reply:
Would it by chance work to use `<P: crate::path::Arg, Fd: AsFd>`?

Reply:
I think it would complain with a missing generic and you'd have to put an underscore. Maybe we could do `impl AsFd` and put a todo to make it a generic later?

Reply:
Making it `impl AsFd` and adding a TODO sounds reasonable to me.

Reply:
Sounds good, done!